django/core/checks/templates.py

```diff
@@ -33,3 +41,32 @@ def check_string_if_invalid_is_string(app_configs, **kwargs):
             error.msg = error.msg.format(string_if_invalid, type(string_if_invalid).__name__)
             errors.append(error)
     return errors
+
+
+@register(Tags.templates)
+def check_for_mulitple_libraries_with_the_same_name(app_configs, **kwargs):
+    errors = []
+
+    libraries = defaultdict(list)
+    candidates = ['django.templatetags']
+    candidates.extend(f'{app.name}.templatetags' for app in apps.get_app_configs())
+
+    for candidate in candidates:
+        try:
+            pkg = import_module(candidate)
+        except ModuleNotFoundError:
+            # No templatetags package defined. This is safe to ignore.
+            continue
```

Comment:
I'd add a new hook to the `django.template.backends.django` that can be used here and in the `get_installed_libraries`, e.g. `get_template_tag_modules()`:
```python
def get_template_tag_modules():
    candidates = ['django.templatetags']
    candidates.extend(f'{app.name}.templatetags' for app in apps.get_app_configs())
    for candidate in candidates:
        try:
            pkg = import_module(candidate)
        except ModuleNotFoundError:
            # No templatetags package defined. This is safe to ignore.
            continue
        if hasattr(pkg, '__path__'):
            for name in get_package_libraries(pkg):
                yield name[len(candidate) + 1:], name
```

Reply:
Good idea!
I'd thought to do the minimal change of just adding the check rather than refactoring anything - but this does make sense!
😄

Reply:
You can add an extra hook in a separate commit.